test/sqlite/result_helper.cpp

```diff
@@ -12,6 +12,8 @@
 #include <thread>
 
 namespace duckdb {
+// we need this shared among all types of result checks
+static std::set<std::string> reported_files;
```

Comment:
Can we keep everything in the `FailureSummary` class instead of introducing more globals?